gnosis/safe/safe_tx.py

```diff
@@ -138,62 +139,67 @@ def safe_version(self) -> str:
             return self.contract.functions.VERSION().call()
 
     @property
-    def eip712_structured_data(self) -> Dict:
+    def _eip712_payload(self) -> StructTuple:
         data = self.data.hex() if self.data else ""
-
-        # Safes >= 1.0.0 Renamed `baseGas` to `dataGas`
         safe_version = Version(self.safe_version)
-        base_gas_name = "baseGas" if safe_version >= Version("1.0.0") else "dataGas"
-
-        structured_data = {
-            "types": {
-                "EIP712Domain": [
-                    {"name": "verifyingContract", "type": "address"},
-                ],
-                "SafeTx": [
-                    {"name": "to", "type": "address"},
-                    {"name": "value", "type": "uint256"},
-                    {"name": "data", "type": "bytes"},
-                    {"name": "operation", "type": "uint8"},
-                    {"name": "safeTxGas", "type": "uint256"},
-                    {"name": base_gas_name, "type": "uint256"},
-                    {"name": "gasPrice", "type": "uint256"},
-                    {"name": "gasToken", "type": "address"},
-                    {"name": "refundReceiver", "type": "address"},
-                    {"name": "nonce", "type": "uint256"},
-                ],
-            },
-            "primaryType": "SafeTx",
-            "domain": {
-                "verifyingContract": self.safe_address,
-            },
-            "message": {
-                "to": self.to,
-                "value": self.value,
-                "data": data,
-                "operation": self.operation,
-                "safeTxGas": self.safe_tx_gas,
-                base_gas_name: self.base_gas,
-                "gasPrice": self.gas_price,
-                "gasToken": self.gas_token,
-                "refundReceiver": self.refund_receiver,
-                "nonce": self.safe_nonce,
-            },
-        }
 
-        # Safes >= 1.3.0 Added `chainId` to the domain
-        if safe_version >= Version("1.3.0"):
-            # EIP712Domain(uint256 chainId,address verifyingContract)
-            structured_data["types"]["EIP712Domain"].insert(
-                0, {"name": "chainId", "type": "uint256"}
-            )
-            structured_data["domain"]["chainId"] = self.chain_id
+        if safe_version >= Version("1.0.0"):
+
+            class SafeTx(EIP712Struct):
```

Comment:
I wouldn't define this class in an `if`, I think it's better to do it at the beginning of the `safe_tx.py` file. Then define 2 classes, `LegacySafeTx` and `SafeTx`

Reply:
Makes sense, pushed this change.